### ISSUE TITLE
Release InfiniDepth artifacts (models, dataset) on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [zju3dv/InfiniDepth#2]
Hi @RitianYu 🤗

Niels here from the open-source team at Hugging Face. I discovered your work through Hugging Face's daily papers as yours got featured: https://huggingface.co/papers/2601.03252.
The paper page lets people discuss about your paper and lets them find artifacts about it (your models, datasets or demo for instance), you can also claim
the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

I saw from your GitHub repository (https://github.com/zju3dv/InfiniDepth) that you have an "Open Source Roadmap" and are planning to release the InfiniDepth model weights (Metric Depth Model, Relative Depth Model) and the Synth4K benchmark dataset. That's fantastic!

It'd be great to make these checkpoints and the dataset available on the 🤗 hub, to improve their discoverability/visibility once they are ready!
We can add tags so that people find them when filtering https://huggingface.co/models and https://huggingface.co/datasets.

## Uploading models

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverages the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

We encourage researchers to push each model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.

## Uploading dataset

Would be awesome to make the dataset available on 🤗 , so that people can do:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/your-dataset")
```
See here for a guide: https://huggingface.co/docs/datasets/loading.

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the first few rows of the data in the browser.

Let me know if you're interested/need any help regarding this!

Cheers,

Niels
ML Engineer @ HF 🤗

**Comment by RitianYu** [comment]
Hi Niels,

Thank you for your interest in our work and for your valuable guidance! We are currently preparing the code, model weights and datasets for public release, and we plan to follow the roadmap in our repository to release them in the coming months.

We really appreciate Hugging Face for offering such a great platform to share and explore research artifacts, and we’d be happy to host our models and datasets on the 🤗 Hub when they’re ready.

Thanks again!

**Comment by pristinaai** [comment]
This is a research game changer. Opening many doors, we are super excited to see what you guys have been working on!

**Comment by NielsRogge** [comment]
Hey @RitianYu, it's great to see that you've uploaded the artifacts to the Hugging Face hub.

However, it seems they are not yet linked to the paper page. Could you link them by mentioning the paper URL (preferably https://huggingface.co/papers/2601.03252, otherwise the arXiv abstract or PDF URL) in the README of the Hugging Face Space repo (https://huggingface.co/spaces/ritianyu/InfiniDepth) and dataset repo (https://huggingface.co/datasets/ritianyu/game_4k_data)?

This will make the artifacts automatically show up on the paper page.

See also here for more information: https://huggingface.co/docs/hub/model-cards#linking-a-paper (applies to Spaces and datasets as well)

Thanks
